src/kernel/multitasking/tasks/task_small.c

```diff
@@ -39,136 +46,198 @@ static task_small_t* _tasking_get_next_task(task_small_t* previous_task) {
     return next_task;
 }
 
+static task_small_t* _tasking_get_next_runnable_task(task_small_t* previous_task) {
+    task_small_t* iter = _tasking_get_next_task(previous_task);
+    for (int i = 0; i < MAX_TASKS; i++) {
+        if (iter->blocked_info.status != RUNNABLE) {
+            iter = _tasking_get_next_task(iter);
+            continue;
+        }
+        return iter;
+    }
+    panic("couldn't find runnable task");
+}
+
 static task_small_t* _tasking_last_task_in_runlist() {
-    if (!_current_task) {
+    if (!_current_task_small) {
         return NULL;
     }
-    task_small_t* iter = _current_task;
-    for (int i = 0; i < 16; i++) {
+    task_small_t* iter = _current_task_small;
+    for (int i = 0; i < MAX_TASKS; i++) {
         if ((iter)->next == NULL) {
             return iter;
         }
         iter = (iter)->next;
     }
-    panic("more than 16 tasks in runlist. increase me?");
+    panic("more than MAX_TASKS tasks in runlist. increase me?");
```

Comment:
Does `panic` take varargs to be able to print the value of `MAX_TASKS`?

Reply:
Unfortunately not, though it should in the future.

Reply:
How about `panic("more than MAX_TASKS(" STR(MAX_TASKS) ") tasks in runlist. increase me?");` where `STR` expands the value of `MAX_TASKS` and then stringifies that with `#`?